common/djangoapps/entitlements/models.py

```diff
@@ -95,8 +96,11 @@ def is_entitlement_regainable(self, entitlement):
             return False
 
         if entitlement.enrollment_course_run:
-            if GeneratedCertificate.certificate_for_student(
-                    entitlement.user_id, entitlement.enrollment_course_run.course_id) is not None:
+            certificate = GeneratedCertificate.certificate_for_student(
```

Comment:
I think we want to move away from using other Django apps models directly wherever possible. Would this be an easy-ish fix while we are modifying this code?